pkg/controllers/metrics/pod/controller.go

```diff
@@ -132,22 +164,57 @@ func (c *Controller) Reconcile(ctx context.Context, req reconcile.Request) (reco
 			Labels:   labels,
 		},
 	})
-	c.recordPodStartupMetric(pod)
+	c.recordPodStartupMetric(pod, labels)
 	return reconcile.Result{}, nil
 }
 
-func (c *Controller) recordPodStartupMetric(pod *corev1.Pod) {
+func (c *Controller) recordPodStartupMetric(pod *corev1.Pod, labels prometheus.Labels) {
 	key := client.ObjectKeyFromObject(pod).String()
+	condScheduled, ok := lo.Find(pod.Status.Conditions, func(c corev1.PodCondition) bool {
+		return c.Type == corev1.PodScheduled
+	})
 	if pod.Status.Phase == phasePending {
+		if ok && condScheduled.Status != corev1.ConditionTrue {
+			podCurrentUnboundTimeSeconds.With(map[string]string{
+				podName:      pod.Name,
+				podNameSpace: pod.Namespace,
+			}).Set(time.Since(pod.CreationTimestamp.Time).Seconds())
+		}
+		podUnstartedTimeSeconds.With(map[string]string{
+			podName:      pod.Name,
+			podNameSpace: pod.Namespace,
+		}).Set(time.Since(pod.CreationTimestamp.Time).Seconds())
 		c.pendingPods.Insert(key)
+		c.unscheduledPods.Insert(key)
 		return
 	}
-	cond, ok := lo.Find(pod.Status.Conditions, func(c corev1.PodCondition) bool {
+	if c.unscheduledPods.Has(key) && ok && condScheduled.Status == corev1.ConditionTrue {
+		// Delete the unbound metric since the pod is now bound
+		podCurrentUnboundTimeSeconds.Delete(map[string]string{
+			podName:      pod.Name,
+			podNameSpace: pod.Namespace,
+		})
+		podBoundDurationSeconds.With(labels).Observe(condScheduled.LastTransitionTime.Sub(pod.CreationTimestamp.Time).Seconds())
+		c.unscheduledPods.Delete(key)
+	}
+	condReady, ok := lo.Find(pod.Status.Conditions, func(c corev1.PodCondition) bool {
 		return c.Type == corev1.PodReady
 	})
 	if c.pendingPods.Has(key) && ok {
```

Comment:
I'm a little worried about how complicated this function is getting and the number of conditional paths and data structures that we have to reason about. If this function gets broken up I think that it might help with being able to reason about everything